### ISSUE TITLE
Link check treats trailing backtick as part of URL when URL is inside code span

### ISSUE THREAD

**Issue by philip** [agent-ecosystem/skill-validator#15]
When a bare URL appears inside a markdown code span, the link extractor includes the closing backtick in the URL, so the checker reports a 404 for a valid link.

**Example**

```markdown
Use: `curl -H "Accept: text/markdown" https://example.com/docs/guide`
```

Current: checker fetches `https://example.com/docs/guide` + trailing backtick -> HTTP 404.  
Expected: URL parsed as `https://example.com/docs/guide` (no backtick).

**Cause**

In `internal/links/links.go`, `bareURLPattern` uses a URL character class that does not exclude backticks, so the closing `` ` `` of a code span is captured as part of the URL.

**Possible direction for fix**

Align with how autolinks are delimited in common markdown (e.g. cmark-gfm’s `autolink_delim`) so trailing punctuation/delimiters are not included.
